pinot-core/src/main/java/org/apache/pinot/core/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -115,7 +116,11 @@ public DataSource getDataSource(String column) {
     ColumnMetadata columnMetadata = _segmentMetadata.getColumnMetadataFor(column);
     Preconditions.checkNotNull(columnMetadata,
         "ColumnMetadata for " + column + " should not be null. " + "Potentially invalid column name specified.");
-    return new ImmutableDataSource(columnMetadata, _indexContainerMap.get(column));
+    if (_indexContainerMap.containsKey(column)) {
```

Comment:
This is not needed

Reply:
When would we use EmptySegmentDataSource in that case?

Reply:
Nvm, I understood what you meant